docs/dotnet.rst

```diff
@@ -0,0 +1,134 @@
+
+3.  Create a new folder somewhere different from your existing project
+
+4.  Create a YAML file titled ``<app-id>.yaml`` with the following example template, replacing the placeholders with the appropriate information. (Note: If your project file lives in a subfolder, be sure to include that in the build paths in this file and subsequent commands as well.): 
+
+.. code-block:: yaml
+
+  id: <app-id>
+  runtime: org.freedesktop.Platform
+  runtime-version: '23.08'
+  sdk: org.freedesktop.Sdk
+  sdk-extensions:
+    - org.freedesktop.Sdk.Extension.dotnet8
+  build-options:
+    prepend-path: "/usr/lib/sdk/dotnet8/bin"
+    append-ld-library-path: "/usr/lib/sdk/dotnet8/lib"
+    prepend-pkg-config-path: "/usr/lib/sdk/dotnet8/lib/pkgconfig"
+
+  command: <project-name>
+
+  finish-args:
+    - --device=dri
+    # TODO: Replace this with wayland and fallback-x11 once Wayland support
+    #       becomes available:
+    #       https://github.com/AvaloniaUI/Avalonia/pull/8003
+    - --socket=x11
+    - --share=ipc
+    - --env=DOTNET_ROOT=/app/lib/dotnet
+
+  modules:
+    - name: dotnet
+      buildsystem: simple
+      build-commands:
+      - /usr/lib/sdk/dotnet8/bin/install.sh
+
+    - name: <app-name>
+      buildsystem: simple
+      sources:
+        - type: git
+          url: <git-url>
+          tag: <git-tag>
+        - ./nuget-sources.json
+      build-commands:
+        - dotnet publish <project-name>.csproj -c Release --no-self-contained --source ./nuget-sources
+        - mkdir -p ${FLATPAK_DEST}/bin
+        - cp -r bin/Release/net8.0/publish/* ${FLATPAK_DEST}/bin
+
+.. note::
+
+    For providing access to other things such as the network or
+    filesystem, see :doc:`sandbox-permissions`
+
+5.  Copy and save the dotnet NuGet sources generator script
+    ``flatpak-dotnet-generator.py`` from the `Flatpak Builder Tools
+    repository <https://github.com/flatpak/flatpak-builder-tools>`_, to
+    the current folder, or run the following command to download it:
+
+.. code-block:: shell
+
+      wget https://raw.githubusercontent.com/flatpak/flatpak-builder-tools/master/dotnet/flatpak-dotnet-generator.py
+
+6.  Clone down your project repository to the folder
+
+.. code-block:: shell
+
+      git clone <git-url>
+
+7.  Install dependencies from Flathub
+
+.. code-block:: shell
+
+      flatpak-builder build-dir --user --repo=repo --install-deps-from=flathub <app-id>.yaml
+
+8.  Run the NuGet source config generator script ``flatpak-dotnet-generator.py`` with the following arguments:
+
+.. code-block:: shell
+
+      python3 flatpak-dotnet-generator.py --dotnet 8 --freedesktop 23.08 nuget-sources.json <app-name>/<project-name>.csproj
+
+9. Run the Flatpak Builder script to build and install the local Flatpak
+
+.. code-block:: shell
+
+      flatpak-builder build-dir --user --force-clean --install --repo=repo <app-id>.yaml
```

Comment:
```suggestion
      flatpak-builder build-dir --user --force-clean --install --repo=repo --install-deps-from=flathub <app-id>.yaml
```

Reply:
(This one may be redundant if the changes referenced in comment [here](https://github.com/flatpak/flatpak-docs/pull/408#discussion_r1689040817) should stay as it is)

Reply:
OK